libnavigator/src/androidTest/java/com/mapbox/navigation/navigator/internal/NavigatorTest.kt

```diff
@@ -0,0 +1,128 @@
+package com.mapbox.navigation.navigator.internal
+
+import androidx.annotation.RawRes
+import androidx.test.espresso.Espresso
+import androidx.test.platform.app.InstrumentationRegistry
+import com.mapbox.api.directions.v5.DirectionsCriteria
+import com.mapbox.api.directions.v5.models.DirectionsResponse
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.navigator.internal.util.readRawFileText
+import com.mapbox.navigation.navigator.internal.util.runOnMainSync
+import com.mapbox.navigator.CacheFactory
+import com.mapbox.navigator.CacheHandle
+import com.mapbox.navigator.ConfigFactory
+import com.mapbox.navigator.ConfigHandle
+import com.mapbox.navigator.Navigator
+import com.mapbox.navigator.NavigatorConfig
+import com.mapbox.navigator.ProfileApplication
+import com.mapbox.navigator.ProfilePlatform
+import com.mapbox.navigator.Routes
+import com.mapbox.navigator.SettingsProfile
+import com.mapbox.navigator.TilesConfig
+import org.junit.Assert.assertTrue
+import org.junit.Test
+
+class NavigatorTest {
+
+    private companion object {
+        @RawRes
+        private val DIRECTIONS_RESPONSE =
+            com.mapbox.navigation.navigator.test.R.raw.directions_response_two_leg_route
+    }
+
+    @Test(timeout = 10_000)
+    fun successfulSetRoutesToNN() {
+        runOnMainSync {
+            val navigator = provideNavigator()
+            val routes = provideDirectionsRouteAndRouteOptions()
+            navigator.setRoutes(
+                Routes(
+                    routes.mapToDirectionsResponse().toJson(),
+                    0,
+                    0,
+                    routes.first().routeOptions.toUrl("pk.**").toString()
+                )
+            ) { expected ->
+                assertTrue(expected.isValue)
+            }
+        }
+        Espresso.onIdle()
+    }
```

Comment:
The test passes if `Navigator` doesn't call fallback. I suggest make test fail if navigator doesn't call callback
```suggestion
        fun successfulSetRoutesToNN() {
        val setRouteResult = runBlocking(Dispatchers.Main) {
            suspendCoroutine<Expected<String, RouteInfo>> { continuation ->
                val navigator = provideNavigator()
                val routes = provideDirectionsRouteAndRouteOptions()
                navigator.setRoutes(
                    Routes(
                        routes.mapToDirectionsResponse().toJson(),
                        0,
                        0,
                        routes.first().routeOptions.toUrl("pk.**").toString()
                    )
                ) { result ->
                    continuation.resume(result)
                }
            }
        }
        assertTrue(setRouteResult.isValue)
    }

```

Reply:
It makes sense to check if a callback is called 👍 I don't really want to bring coroutines to only check that callback is called. I would make it in a plain way

Reply:
> I don't really want to bring coroutines

Why? Which disadvantages do you see in using coroutines here?